ts/tests/unit_tests/metrics_yaml_testing/metric_cache_unit_test.py

```diff
@@ -0,0 +1,1270 @@
+            uuid.uuid4(), "ModelNameExample", "metrics.yaml"
+        )
+        assert isinstance(metrics_cache_obj._parse_specific_metric(), dict)
+
+    def test_pass_parse_ts_metrics(self):
+        metrics_cache_obj = MetricsCacheYaml(
+            uuid.uuid4(), "ModelNameExample", "metrics.yaml"
+        )
+        assert isinstance(metrics_cache_obj._parse_specific_metric("ts_metrics"), dict)
+
+    def test_fail_parse_model_metrics(self):
+        metrics_cache_obj = MetricsCacheYaml(
+            uuid.uuid4(), "ModelNameExample", "metrics_wo_model_metrics.yaml"
+        )
+        with pytest.raises(merrors.MetricsCacheKeyError) as exc_info:
+            metrics_cache_obj._parse_specific_metric()
+        assert (
+            str(exc_info)
+            == "<ExceptionInfo MetricsCacheKeyError(\"'model_metrics' key not found in yaml "
+            "file - 'model_metrics'\") tblen=2>"
+        )
+
+    def test_fail_parse_none(self):
+        metrics_cache_obj = MetricsCacheYaml(
+            uuid.uuid4(), "ModelNameExample", "metrics_wo_model_metrics.yaml"
+        )
+        with pytest.raises(merrors.MetricsCacheKeyError) as exc_info:
+            metrics_cache_obj._parse_specific_metric(None)
+        assert (
+            str(exc_info)
+            == "<ExceptionInfo MetricsCacheKeyError(\"'None' key not found in yaml file - "
+            'None") tblen=2>'
+        )
+
+    def test_fail_parse_missing_fields(self):
+        metrics_cache_obj = MetricsCacheYaml(
+            uuid.uuid4(), "ModelNameExample", "metrics_model_empty.yaml"
+        )
+        assert metrics_cache_obj._parse_specific_metric() is None
+
+
+class TestYamlCacheUtil:
+    def test_yaml_to_cache_util_pass(self):
+        metrics_cache_obj = MetricsCacheYaml(
+            uuid.uuid4(), "ModelNameExample", "metrics.yaml"
+        )
+        model_metrics_table = metrics_cache_obj._parse_specific_metric()
+        metrics_cache_obj._yaml_to_cache_util(model_metrics_table)
+        assert list(metrics_cache_obj.cache.keys()) == [
+            "[counter]-[InferenceTimeInMS]-[ModelName:my_tc,Level:Model]",
+            "[counter]-[NumberOfMetrics]-[model_name:example_model_name,host:example_host_name"
+            ",ModelName:ModelNameExample,Level:Model]",
+            "[gauge]-[GaugeModelMetricNameExample]-[ModelName:my_tc,Level:Model]",
+            "[histogram]-[HistogramModelMetricNameExample]-[ModelName:my_tc,Level:Model]",
+        ]
+
+    def test_yaml_to_cache_empty_dims(self):
+        metrics_cache_obj = MetricsCacheYaml(
+            uuid.uuid4(), "ModelNameExample", "metrics_empty_dims.yml"
+        )
+        model_metrics_table = metrics_cache_obj._parse_specific_metric()
+        print(model_metrics_table)
+        metrics_cache_obj._yaml_to_cache_util(model_metrics_table)
+        assert list(metrics_cache_obj.cache.keys()) == [
+            "[counter]-[InferenceTimeInMS]-[ModelName:ModelNameExample,Level:Model]",
```

Comment:
This seems less than ideal, instead of asserting on the equivalence of strings I'd like to assert on the equivalence of some data structures